src/libraries/Native/Unix/System.Security.Cryptography.Native.Apple/pal_digest.c

```diff
@@ -171,3 +171,32 @@ int32_t AppleCryptoNative_DigestCurrent(const DigestCtx* ctx, uint8_t* pOutput,
     DigestCtx dup = *ctx;
     return AppleCryptoNative_DigestFinal(&dup, pOutput, cbOutput);
 }
+
+int32_t AppleCryptoNative_DigestOneShot(PAL_HashAlgorithm algorithm, uint8_t* pBuf, int32_t cbBuf, uint8_t* pOutput, int32_t cbOutput, int32_t* pcbDigest)
+{
+    if (pOutput == NULL || cbOutput <= 0 || pcbDigest == NULL)
+        return -1;
+
+    DigestCtx* ctx = AppleCryptoNative_DigestCreate(algorithm, pcbDigest);
```

Comment:
Can we not write the one-shot here in terms of (e.g.) `CC_SHA1(const void *data, CC_LONG len, unsigned char *md);`?

Reply:
Huh, yes we can. Will address.